src/tree/FunctionAppProvider.ts

```diff
@@ -83,47 +83,59 @@ export class FunctionAppProvider extends SubscriptionTreeItem {
         promptSteps.push(new SiteRuntimeStep());
-        promptSteps.push(new StorageAccountListStep(
-            {
-                kind: StorageAccountKind.Storage,
-                performance: StorageAccountPerformance.Standard,
-                replication: StorageAccountReplication.LRS
-            },
-            {
-                kind: [
-                    StorageAccountKind.BlobStorage
-                ],
-                performance: [
-                    StorageAccountPerformance.Premium
-                ],
-                replication: [
-                    StorageAccountReplication.ZRS
-                ],
-                learnMoreLink: 'https://aka.ms/Cfqnrc'
-            }
-        ));
 
-        const executeSteps: AzureWizardExecuteStep<IAppServiceWizardContext>[] = [new SiteCreateStep(
-            async (context: IAppSettingsContext): Promise<WebSiteManagementModels.NameValuePair[]> => await createFunctionAppSettings(context, runtime, language)
-        )];
+        const storageAccountCreateOptions: INewStorageAccountDefaults = {
+            kind: StorageAccountKind.Storage,
+            performance: StorageAccountPerformance.Standard,
+            replication: StorageAccountReplication.LRS
+        };
+
+        const advancedCreationKey: string = 'advancedCreation';
+        const advancedCreation: boolean = !!getFuncExtensionSetting(advancedCreationKey);
+        actionContext.properties.advancedCreation = String(advancedCreation);
+        if (!advancedCreation) {
+            await setBasicCreateDefaults(wizardContext, language);
+            executeSteps.push(new ResourceGroupCreateStep());
+            executeSteps.push(new StorageAccountCreateStep(storageAccountCreateOptions));
+        } else {
+            promptSteps.push(new ResourceGroupListStep());
+            promptSteps.push(new StorageAccountListStep(
+                storageAccountCreateOptions,
+                {
+                    kind: [
+                        StorageAccountKind.BlobStorage
+                    ],
+                    performance: [
+                        StorageAccountPerformance.Premium
+                    ],
+                    replication: [
+                        StorageAccountReplication.ZRS
+                    ],
+                    learnMoreLink: 'https://aka.ms/Cfqnrc'
+                }
+            ));
+        }
+
+        executeSteps.push(new SiteCreateStep(async (context): Promise<WebSiteManagementModels.NameValuePair[]> => await createFunctionAppSettings(context, runtime, language)));
 
         const title: string = localize('functionAppCreatingTitle', 'Create new function app');
         const wizard: AzureWizard<IAppServiceWizardContext> = new AzureWizard(wizardContext, { promptSteps, executeSteps, title });
 
         await wizard.prompt(actionContext);
         showCreatingTreeItem(nonNullProp(wizardContext, 'newSiteName'));
+        if (!advancedCreation) {
```

Comment:
How come this can't be done within the if statement from lines 101-104?  Or in `setBasicCreateDefaults`?

Reply:
`wizardContext.relatedNameTask;` is based on the function app name they entered, so it only exists after we call `wizard.prompt`